src/Tools/vmOpsTool/VMWareImpl.java

```diff
@@ -608,11 +608,15 @@ private ManagedObjectReference findSnapshotInTree(List<VirtualMachineSnapshotTre
 
             if (vmSnapshot.getName().equalsIgnoreCase(snapshotName)) {
                 System.out.println(String.format("Found snapshot [ %s ] for virtual machine.", snapshotName));
-                return vmSnapshot.getSnapshot();
+                snapshotMor =  vmSnapshot.getSnapshot();

```

Comment:
It has to be return only otherwise if there are n snap shots in same level then required one is found in n-3. Still the loop will continue till n, which is not we wanted it.